r3bbase/R3BFileSource.cxx

```diff
@@ -461,15 +504,16 @@ Int_t R3BFileSource::ReadEvent(UInt_t i)
      ** std::cout << fEvtHeader->GetTimeStamp() << std::endl;
      **/
 
-    if (nextts == 0 && fEvtHeader->GetTimeStamp() > nextts)
-    {
-        fRunId = getrunid(fEvtHeader->GetTimeStamp());
-        fEvtHeader->SetRunId(fRunId);
-    }
-    else if (nextts > 0 && fEvtHeader->GetTimeStamp() > nextts)
+    printf("Processed: %d of %d (%.2f of 100), current RunId %d \r",
+           i,
+           fNoOfEntries,
+           100. * i / (double)(fNoOfEntries),
+           fRunId);
+    fflush(stdout);
+
+    if (nextts >= 0 && prevts >= 0 && (fEvtHeader->GetTimeStamp() > nextts || fEvtHeader->GetTimeStamp() < prevts))
     {
-        fRunId = getrunid(fEvtHeader->GetTimeStamp());
-        fEvtHeader->SetRunId(fRunId);
+        fRunId = GetRunid(fEvtHeader->GetTimeStamp());
     }
```

Comment:
What should fRunId be set to, if the condition is false?

Reply:
It will take the last value for the fRunId, or if the timestamp is not defined inside the runid file it will be set to 1.